src/main/java/pl/betse/beontime/timesheet/controller/WeekTimeEntryController.java

```diff
@@ -0,0 +1,109 @@
+        Link link = new Link(location.toString(), "self");
+        return ResponseEntity.ok(new Resources<>(weekTimeEntryBodyList, link));
+    }
+
+    @PostMapping("/{userGuid}/week/{weekNumber}")
+    public ResponseEntity createWeekForUser(
+            @RequestBody @Validated(CreateTimeEntry.class) WeekTimeEntryBody weekTimeEntryBody,
+            @PathVariable("userGuid") String userGuid,
+            @PathVariable("weekNumber") String weekNumber,
+            HttpServletRequest httpServletRequest) {
+        List<TimeEntryBo> timeEntryBoList = prepareDataForService(weekTimeEntryBody, userGuid, weekNumber);
+        timeEntryService.checkIfDateIsInCorrectWeekOfYear(timeEntryBoList, weekNumber);
+        timeEntryService.checkIfTimeEntriesExist(timeEntryBoList, httpServletRequest.getMethod());
+        timeEntryService.verifyStatusesBeforeCreatingNewEntry(timeEntryBoList);
+        timeEntryService.saveWeekForUser(timeEntryBoList);
+        URI location = linkTo(methodOn(WeekTimeEntryController.class).getWeekForUser(userGuid, weekNumber)).toUri();
+        return ResponseEntity.created(location).build();
+    }
+
+    @PutMapping("/{userGuid}/week/{weekNumber}")
+    public ResponseEntity editWeekForUser(
+            @RequestBody @Validated(CreateTimeEntry.class) WeekTimeEntryBody weekTimeEntryBody,
+            @PathVariable("userGuid") String userGuid,
+            @PathVariable("weekNumber") String weekNumber,
+            HttpServletRequest httpServletRequest) {
+        List<TimeEntryBo> timeEntryBoList = prepareDataForService(weekTimeEntryBody, userGuid, weekNumber);
+        timeEntryService.checkIfDateIsInCorrectWeekOfYear(timeEntryBoList, weekNumber);
+        timeEntryService.checkIfTimeEntriesExist(timeEntryBoList, httpServletRequest.getMethod());
+        timeEntryService.verifyStatusesBeforeAddingComment(timeEntryBoList);
+        timeEntryService.editWeekHoursAndStatuses(timeEntryBoList, userGuid, weekNumber);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping("/{userGuid}/week/{weekNumber}")
+    public ResponseEntity deleteWeekForUser(@RequestBody @Validated(CreateTimeEntry.class) WeekTimeEntryBody weekTimeEntryBody,
+                                            @PathVariable("userGuid") String userGuid,
+                                            @PathVariable("weekNumber") String weekNumber,
+                                            HttpServletRequest httpServletRequest) {
+        List<TimeEntryBo> timeEntryBoList = prepareDataForService(weekTimeEntryBody, userGuid, weekNumber);
+        timeEntryService.checkIfDateIsInCorrectWeekOfYear(timeEntryBoList, weekNumber);
+        timeEntryService.checkIfTimeEntriesExist(timeEntryBoList, httpServletRequest.getMethod());
+        timeEntryService.verifyStatusesBeforeDeleting(timeEntryBoList);
+        timeEntryService.deleteWholeWeek(timeEntryBoList, userGuid, weekNumber);
```

Comment:
Shall I be able to delete a week??? Each year have around 52 weeks - it is not something I can remove

Reply:
If your entry have status SAVED -> yes you can delete it from database. (but like I mention earlier all operations for user are operated weekly, so it can be quite confusing). 

Any other statuses: SUBMITTED, APPROVED or REJECTED provides that entries cannot be deleted.